zk/stages/stage_hashstate.go

```diff
@@ -0,0 +1,991 @@
+package stages
```

Comment:
We shouldn't need to copy and paste this whole file for a small change.  We can have a new function called something like `HashStateFromTo` and the `SpawnHashStateStage` just looks up the from/to and calls that function.  Then we can use the same code from all places for consistency.